asm-to-pil/src/vm_to_constrained.rs

```diff
@@ -283,10 +283,11 @@ impl<T: FieldElement> VMConverter<T> {
         // This is hacky: in the absence of proof objects, we want to support both monolithic proofs and composite proofs.
         // In the monolithic case, all degrees must be the same, so we align the degree of the rom to that of the vm.
         // In the composite case, we set the minimum degree for the rom, which is the number of lines in the code.
-        let rom_degree = input
-            .degree
-            .clone()
-            .unwrap_or_else(|| Expression::from(self.code_lines.len().next_power_of_two() as u32));
+        // this can lead to false negatives as we apply expression equality here, so `4` and `2 + 2` would be considered different.
```

Comment:
Where do we check for expression equality?

Reply:
Oh, this looks like a leftover comment of something that got moved somewhere else

Reply:
Actually no, we check expression equality in `is_static`